runtime/src/bank/pyth/tests/batch_publish_tests.rs

```diff
@@ -0,0 +1,147 @@
+use {
+    crate::{
+        bank::pyth::{
+            accumulator::{BATCH_PUBLISH_PID, ORACLE_PID},
+            tests::{create_new_bank_for_tests_with_index, new_from_parent},
+        },
+        genesis_utils::{create_genesis_config_with_leader, GenesisConfigInfo},
+    },
+    bytemuck::{cast_slice, checked::from_bytes},
+    pyth_oracle::{
+        solana_program::account_info::AccountInfo, PriceAccount, PriceAccountFlags, PythAccount,
+    },
+    pyth_price_publisher::accounts::publisher_prices::{
+        self as publisher_prices_account, PublisherPrice,
+    },
+    solana_sdk::{
+        account::{AccountSharedData, ReadableAccount, WritableAccount},
+        clock::Epoch,
+        epoch_schedule::EpochSchedule,
+        feature_set,
+        pubkey::Pubkey,
+        signature::keypair_from_seed,
+        signer::Signer,
+    },
+    std::{mem::size_of, sync::Arc},
+};
+
+#[test]
+fn test_batch_publish() {
```

Comment:
if feed index of 0 is permitted we can add tests with that and make sure that it doesn't work.

Reply:
Expanded the test to make sure prices with feed_index == 0 are not applied.